### PR TITLE
feat: add views for webpay controllers when the waiting time is exceeded in the Webpay form

### PR DESCRIPTION
**Pull request by DiegoHBustamante**
This version includes the following changes:

- Add timeout view when the waiting time is exceeded in the Webpay form in webpay plus, webpay plus deferred, webpay plus mall and webpay plus mall deferred.
- Add conditions in WebpayPlusController, WebpayPlusDeferredController, WebpayPlusMallController and WebpayPlusMallDeferredController to control timeout flow.
- Fix Webpay plus aborted view.

These changes were made because the timeout flow in the Webpay form is not controlled.

> Case with modifications
Captura de Pantalla 2024-06-19 a la(s) 19 08 36

### CODE REVIEW

File: src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusController.java [hunk 1]
```diff
@@ -71,17 +71,23 @@ public ModelAndView create(HttpServletRequest request) {
     public ModelAndView commit(HttpServletRequest request) {
         String tokenWs = request.getParameter("token_ws");
         Map<String, Object> details = new HashMap<>();
-
-        if (tokenWs == null) {
+        String buyOrder = (String) request.getSession().getAttribute("buyOrder");
+        String sessionId = (String) request.getSession().getAttribute("sessionId");
+        if(request.getParameter("TBK_TOKEN")==null && tokenWs==null || request.getParameter("TBK_TOKEN").isEmpty() && tokenWs.isEmpty())
```
Comment (tribiec) [inline src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusController.java]: Suggestion: The logic inside the if should be moved to a specific variable, such as transactionHasTimeout or something similar.

File: src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusDeferredController.java [hunk 2]
```diff
@@ -64,11 +64,16 @@ public ModelAndView create(HttpServletRequest request) {
     public ModelAndView commit(HttpServletRequest request) {
         String tokenWs = request.getParameter("token_ws");
         Map<String, Object> details = new HashMap<>();
-
-        if (tokenWs == null) {
+        String buyOrder = (String) request.getSession().getAttribute("buyOrder");
+        String sessionId = (String) request.getSession().getAttribute("sessionId");
+        if(request.getParameter("TBK_TOKEN")==null && tokenWs==null || request.getParameter("TBK_TOKEN").isEmpty() && tokenWs.isEmpty())
```
Comment (tribiec) [inline src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusDeferredController.java]: The same here, move the if logic to a variable

File: src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusMallController.java [hunk 3]
```diff
@@ -82,11 +82,16 @@ public ModelAndView commit(HttpServletRequest request) {
 
         String tokenWs = request.getParameter("token_ws");
         Map<String, Object> details = new HashMap<>();
-
+        String buyOrder = (String) request.getSession().getAttribute("buyOrder");
+        String sessionId = (String) request.getSession().getAttribute("sessionId");
+        if(request.getParameter("TBK_TOKEN")==null && tokenWs==null || request.getParameter("TBK_TOKEN").isEmpty() && tokenWs.isEmpty())
```
Comment (tribiec) [inline src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusMallController.java]: The same here

File: src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusMallDeferredController.java [hunk 4]
```diff
@@ -81,11 +81,17 @@ public ModelAndView commit(HttpServletRequest request) {
 
         String tokenWs = request.getParameter("token_ws");
         Map<String, Object> details = new HashMap<>();
+        String buyOrder = (String) request.getSession().getAttribute("buyOrder");
+        String sessionId = (String) request.getSession().getAttribute("sessionId");
+        if(request.getParameter("TBK_TOKEN")==null && tokenWs==null || request.getParameter("TBK_TOKEN").isEmpty() && tokenWs.isEmpty())
```
Comment (tribiec) [inline src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusMallDeferredController.java]: ^^

File: src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusController.java (hunk 1 above)
Comment (DiegoHBustamante) [reply src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusController.java]: The suggested changes were made

File: src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusDeferredController.java (hunk 2 above)
Comment (DiegoHBustamante) [reply src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusDeferredController.java]: The suggested changes were made

File: src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusMallController.java (hunk 3 above)
Comment (DiegoHBustamante) [reply src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusMallController.java]: The suggested changes were made

File: src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusMallDeferredController.java (hunk 4 above)
Comment (DiegoHBustamante) [reply src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusMallDeferredController.java]: The suggested changes were made

**Review by tribiec — APPROVED** [review]
GJ!

**Review by mastudillot — APPROVED** [review]
LGTM

**Pull request merged**